packages/openactive-integration-tests/test/features/cancellation/seller-requested-cancellation-message/implemented/seller-requested-cancellation-with-message-test.js

```diff
@@ -1,111 +1,97 @@
-      state, flow, logger,
-    }))
-      .beforeSetup()
-      .successChecks()
-      .validationTests();
+  const b = new BFlowStage({
+    ...defaultFlowStageParams,
+    prerequisite: c2,
+    getInput: () => ({
+      orderItems: fetchOpportunities.getOutput().orderItems,
+      totalPaymentDue: c2.getOutput().totalPaymentDue,
+    }),
   });
-
-  describe('Simulate Seller Cancellation (Test Interface Action)', () => {
-    const response = (new TestInterfaceAction({
-      flow,
-      logger,
+  const [simulateSellerCancellation, orderFeedUpdate] = OrderFeedUpdateFlowStageUtils.wrap({
+    wrappedStageFn: prerequisite => (new TestInterfaceActionFlowStage({
+      ...defaultFlowStageParams,
+      testName: 'Simulate Seller Cancellation (Test Interface Action)',
+      prerequisite,
       createActionFn: () => ({
         type: 'test:SellerRequestedCancellationWithMessageSimulateAction',
         objectType: 'Order',
-        objectId: state.bResponse.body['@id'],
+        objectId: b.getOutput().orderId,
       }),
-      completedFlowStage: 'B',
-    }))
-      .beforeSetup()
-      .successChecks();
+    })),
+    orderFeedUpdateParams: {
+      ...defaultFlowStageParams,
+      prerequisite: b,
+      testName: 'Orders Feed (after Simulate Seller Cancellation)',
+    },
   });
 
-  // TODO: Orders feed check
-  //   describe('Orders Feed (after U)', () => {
-  //       const orderFeedUpdate = (new OrderFeedUpdate({
-  //         state,
-  //         flow,
-  //         logger,
-  //         ordersFeedMode: 'orders-feed-after-u',
-  //       }))
-  //         .beforeSetup()
-  //         .successChecks()
-  //         .validationTests();
-
-//     it('should have orderStatus: SellerCancelled', () => {
-//       expect(orderFeedUpdate.getStateResponse().body).to.have.nested.property('data.orderStatus', 'https://openactive.io/SellerCancelled');
-//       expect(orderFeedUpdate.getStateResponse().body).to.have.nested.property('data.cancellationMessage', 'SellerCancelled');
-//     });
-//   });
+  // ## Set up tests
+  FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(fetchOpportunities);
+  FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(c1);
+  FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(c2);
+  FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(b);
+  FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(simulateSellerCancellation);
+  FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(orderFeedUpdate, () => {
+    it('should have orderItemStatus: SellerCancelled', () => {
+      const orderItems = orderFeedUpdate.getOutput().httpResponse.body.data.orderedItem;
+      // As we'll be setting out expectations in an iteration, this test would
+      // give a false positive if there were no items in `orderedItem`, so we
+      // explicitly test that the OrderItems are present.
+      expect(orderItems).to.be.an('array').with.lengthOf(fetchOpportunities.getOutput().orderItems.length);
```

Comment:
```suggestion
      expect(orderItems).to.be.an('array').with.lengthOf(orderItemCriteriaList.length);
```

Probably should check this against the expectations of the test rather than the previous output?